crates/node-core/src/version.rs

```diff
@@ -11,6 +17,39 @@ pub const VERGEN_GIT_SHA: &str = env!("VERGEN_GIT_SHA");
 /// The build timestamp.
 pub const VERGEN_BUILD_TIMESTAMP: &str = env!("VERGEN_BUILD_TIMESTAMP");
 
+//The identification of the client
+/// - The two letter client code
+/// - The human readable name of the client
+/// - The version string of the current implementation
+/// - first four bytes of the latest commit hash of this build
+///
+/// # Example
+///
+///  ```text
+// {
+//   code: "RH",
+//   name: "Reth",
+//   version: "v0.2.0-beta.5",
+//   commit: "defa64b2"
+// }
+///  ```
+
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub struct ClientVersionV1<'a> {
+    pub code: &'a str,
+    pub name: &'a str,
+    pub version: &'a str,
+    pub commit: &'a str,
+}
```

Comment:
this type already exists somewhere, so don't redefine it, just import it

Reply:
Hey @onbjerg after i import ClientVersionV1 and change the values, i am getting this error  
```
error[E0015]: cannot call non-const fn `<str as ToString>::to_string` in constants
  --> crates/node-core/src/version.rs:39:23
   |
39 |     name: NAME_CLIENT.to_string(),
   |                       ^^^^^^^^^^^
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants

error[E0015]: cannot call non-const fn `<str as ToString>::to_string` in constants
  --> crates/node-core/src/version.rs:40:89
   |
40 |     version: const_str::concat!(env!("CARGO_PKG_VERSION"), env!("RETH_VERSION_SUFFIX")).to_string(),
   |                                                                                         ^^^^^^^^^^^
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants

error[E0015]: cannot call non-const fn `<str as ToString>::to_string` in constants
  --> crates/node-core/src/version.rs:41:36
   |
41 |     commit: env!("VERGEN_GIT_SHA").to_string(),
   |                                    ^^^^^^^^^^^
   |
   = note: calls in constants are limited to constant functions, tuple structs and tuple variants
   ```
   pretty much confused on why this is happening?

Reply:
The error says it: since this is in a constant, you can only call functions marked as `const`, which `ToString::to_string` is not

I'd probably just construct the `ClientVersionV1` when we initialize the engine API and pass that by value, instead of having a constant definition of it